code/dce-downloader/node/dce-common.js

```diff
@@ -0,0 +1,132 @@
+
+const request = require('request-promise');
+const crypto = require('crypto');
+const util = require('util');
+const pathLib = require('path');
+const fs = require('fs');
+
+module.exports = {
+    readConfig : function readConfig(config) {
+        try {
+            return JSON.parse(fs.readFileSync(config, 'utf8'));
+        } catch (e) {
+            console.error("Error reading %s: %s", config, e.toString());
+            process.exit(1);
+        }
+    },
+
```

Comment:
Im not sure if version lower than Node 6 support the syntax but this whole file could be written in a more modern style.

`export function readConfig () { ... }` instead of the older `module.exports` syntax.